core/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/list/ListTasksResponse.java

```diff
@@ -96,6 +100,41 @@ public void writeTo(StreamOutput out) throws IOException {
         return nodeTasks;
     }
 
+    public List<TaskGroup> getTaskGroups() {
+        if (groups == null) {
+            buildTaskGroups();
+        }
+        return groups;
+    }
+
+    private void buildTaskGroups() {
+        Map<TaskId, TaskGroup.Builder> taskGroups = new HashMap<>();
```

Comment:
Maybe a TreeMap if you are just going to build it an then iterate over it. Probably doesn't matter.